src/main/java/ca/corefacility/bioinformatics/irida/processing/impl/FastqcFileProcessor.java

```
@@ -95,44 +98,62 @@ private void processSingleFile(SequenceFile sequenceFile) throws FileProcessorEx
+				handleDuplicationLevel(overRep.duplicationLevelModule(), analysis, outputDirectory);
+				Set<OverrepresentedSequence> overrepresentedSequences = handleOverRepresentedSequences(overRep);
 
-			handleBasicStats(basicStats, analysis);
-			handlePerBaseQualityScores(pbqs, analysis, outputDirectory);
-			handlePerSequenceQualityScores(psqs, analysis, outputDirectory);
-			handleDuplicationLevel(overRep.duplicationLevelModule(), analysis, outputDirectory);
-			Set<OverrepresentedSequence> overrepresentedSequences = handleOverRepresentedSequences(overRep);
+				logger.trace("Saving FastQC analysis.");
+				analysis.overrepresentedSequences(overrepresentedSequences);
 
-			logger.trace("Saving FastQC analysis.");
-			analysis.overrepresentedSequences(overrepresentedSequences);
+				AnalysisFastQC analysisFastQC = analysis.build();
 
-			AnalysisFastQC analysisFastQC = analysis.build();
+				sequenceFile.setFastQCAnalysis(analysis.build());
 
-			sequenceFile.setFastQCAnalysis(analysis.build());
-
-			sequenceFileRepository.saveMetadata(sequenceFile);
-		} catch (Exception e) {
-			logger.error("FastQC failed to process the sequence file: " + e.getMessage());
-			throw new FileProcessorException("FastQC failed to parse the sequence file.", e);
+				sequenceFileRepository.saveMetadata(sequenceFile);
+			} catch (Exception e) {
+				logger.error("FastQC failed to process the sequence file: " + e.getMessage());
+				throw new FileProcessorException("FastQC failed to parse the sequence file.", e);
+			}
+		} catch (IOException e) {
+			logger.error("Unable to create temporary directory ", e);
+			throw new StorageException("Unable to create temporary directory", e);
+		} finally {
+				iridaFileStorageUtility.cleanupDownloadedLocalTemporaryFiles(iridaTemporaryFile);
+				try {
+					// Delete the analysis-output* temp directory
+					Files.deleteIfExists(outputDirectory);
+				} catch (IOException e) {
+					logger.error("Unable to remove directory", e);
```

Should this exception be wrapped and re-thrown as a `StorageException` or `FileProcessorException` rather than just logging?

---

Updated in 117eedc7f06e586f08eed0fca9d520de62ca3c6f